### DISCUSSION TITLE
Additions for the OpenGL 3.3 shaders

### DISCUSSION THREAD

**Discussion by Fenix770** [86Box/86Box#4261] [Ideas]
**Is your feature request related to a problem? Please describe.**

Feature request:

1. I'd would be intresting if configuration options were added for the OpenGL shaders, similar to those implemented on PCEM.

2. Add compatibility for Intel HD 4400: My machine has both the Intel HD and Nvidia 820M graphics chip. Only the Nvidia works on 86box. On PCEM both chips works. Also the Intel  works on IBMulator which also uses OpenGL 3.3. 
Also HD4400 supports OpenGL ver 4.3 and the 820M OpenGL 4.6.

Screenshoths:

![Shaders configuration](https://user-images.githubusercontent.com/31600563/116167571-dc0d3f80-a6d6-11eb-94fb-1ff3fa95ff57.png)

**Additional context**
In general the 820M is faster than the HD4400, but on PCEM the shaders uses about 25%-27% of GPU load on any of them.

**Comment by OBattler** [comment]
Note: If possible, add CGP support as well (see issue #4073).

**Comment by chroniceel93** [comment]
Just a side note from the peanut gallery- Libretro's Shader handling code *is* MIT Licensed, which means it's more than possible to port it over, and integrate here as-is.
The Pros to this would be more-or-less perfect compatibility with libretro shaders
The obvious drawback would be maintenance, and whatever effort would be needed to actually integrate things. 86Box already has a GLSL backend, and functionality to select .glsl shaders so one would assume, somewhat naively, that most of the work has already been done.
